client/common/cmdline.c

```diff
@@ -795,27 +795,61 @@ int freerdp_client_command_line_post_filter(void* context, COMMAND_LINE_ARGUMENT
 int freerdp_parse_username(char* username, char** user, char** domain)
 {
 	char* p;
-	int length;
+	char* u;
+	int length = 0;
 
 	p = strchr(username, '\\');
+	p = strrchr(username, '@');
+
+	*user = NULL;
+	*domain = NULL;
 
 	if (p)
 	{
 		length = (int) (p - username);
+		*user = _strdup(&p[1]);
+		if (!*user)
+			return -1;
+
 		*domain = (char*) calloc(length + 1UL, sizeof(char));
+		if (!*domain)
+		{
+			free (*user);
+			*user = NULL;
+			return -1;
+		}
+
 		strncpy(*domain, username, length);
 		(*domain)[length] = '\0';
-		*user = _strdup(&p[1]);
 	}
-	else
+	else if (username)
 	{
 		/* Do not break up the name for '@'; both credSSP and the
 		 * ClientInfo PDU expect 'user@corp.net' to be transmitted
-		 * as username 'user@corp.net', domain empty.
+		 * as username 'user@corp.net', domain empty (not NULL!).
 		 */
 		*user = _strdup(username);
-		*domain = NULL;
+		if (!*user)
+			return -1;
+
+		/* If only username is given, prefix that with 'TARGET'
+		 * otherwise set the domain to an empty string.
+		 * NOTE: Domain NULL will result in undefined behavior.
+		 */
+		if (!u)
+			*domain = _strdup("TARGET");
```

Comment:
Although it seems to work this feels really weird. This would mean that you can't use TARGET as domain name.
@awakecoding do you know something about that or have any ideas?